src/vehicle.cpp

```diff
@@ -1636,6 +1636,101 @@ bool vehicle::remove_carried_vehicle( std::vector<int> carried_parts )
     return success;
 }
 
+// split the current vehicle into up to 3 new vehicles that do not connect to each other
+bool vehicle::find_and_split_vehicles( int exclude )
+{
+    std::vector<int> valid_parts = all_parts_at_location( part_location_structure );
+    std::set<int> checked_parts;
+    checked_parts.insert( exclude );
+
+    std::vector<std::vector <int>> all_vehicles;
+
+    size_t cnt;
```

Comment:
Whydid you define this outside of the loop (it's only used inside of it)?

Reply:
Fixed.